### PR TITLE
Fix sample viewer API issues from holistic testing, add tooltip

### PR DESCRIPTION
**Pull request by Jade-JadeH**
A couple of small UX fixes: 
- [x] Fixed an known issue that enter API key in project settings doesn't unlock the samples reported by holistic sample users 
- [x] fixed the issue that api key in project settings doesn't allow services such as routing and geocoding 
- [x] upon discussing with Andy, instead of using banner, use a warning sign and tooltip to remind user to enter API key
- [x] Make the warning sign comes and go as users enter and remove API keys in the text box (project settings doesn't update in runtime though)

### CODE REVIEW

**Review by mmgaw113 — COMMENTED** [review]
Looks good. I'll approve after you deal with merge conflicts and then I will approve.

**Review by avr757 — CHANGES_REQUESTED** [review]
Looks good and the tooltip works great! 

Just one small change - for the Note, the gap between the title and paragraph can be reduced a little bit (may need to shorten the menu vertically) when the first sentence "Don't forget to enter your API key" is removed from the panel. (just to align with other Notes for consistency purposes).

Current one (more gap between the title and paragraph):
image


Notes from other samples (less gap between the title and paragraph):
image

**Comment by Jade-JadeH** [comment]
> Just one small change - for the Note, the gap between the title and paragraph can be reduced a little bit (may need to shorten the menu vertically) when the first sentence "Don't forget to enter your API key" is removed from the panel. (just to align with other Notes for consistency purposes).

Good call, I also noticed it but forgot to change

**Review by avr757 — APPROVED** [review]

**Review by avr757 — APPROVED** [review]

**Review by mmgaw113 — APPROVED** [review]

**Pull request merged**
